lib/kdtree.cpp

```diff
@@ -597,12 +596,10 @@ KDTreeIntersection::intersect(const detail::FlatNode* node, const Ray& ray,
 
     for (; node->is_leaf(); ++node) {
```

Comment:
vector is always cuntigious in memory, this is a requirement. I measured the time, and it is faster just to do pointer arithmetic here, than using the subscript operator [] of the vector.